src/server/util/insertData.js

```diff
@@ -453,65 +453,80 @@ async function insertMeters(metersToInsert, conn) {
  * @param {*} conn database connection
  */
 async function insertGroups(groupsToInsert, conn) {
+	// Check that needed keys are there.
+	const requiredKeys = ['name', 'displayable', 'childMeters', 'childGroups'];
 	// We don't use Promise.all since one group may include another group.
 	for (let i = 0; i < groupsToInsert.length; ++i) {
-		// Group values from above.
+		// Group currently working on
 		const groupData = groupsToInsert[i];
-		const groupName = groupData[0];
-		console.log(`            creating group ${groupName}`);
-		// We get the needed unit id from the name given.
-		let groupDefaultGraphicUnit;
-		if (groupData[1] === '') {
-			// No unit so make it -99.
-			groupDefaultGraphicUnit = -99;
-		} else {
-			groupDefaultGraphicUnit = (await Unit.getByName(groupData[1], conn)).id;
-		}
-		const group = new Group(
-			undefined, // id
-			groupName, // name
-			groupData[2], //displayable
-			groupData[3], // gps
-			groupData[4], // note
-			undefined, //area
-			groupDefaultGraphicUnit, // default graphic unit
-			undefined // area unit
-		);
-		if (await group.existsByName(conn)) {
-			console.log(`              Warning: group '${group.name}' existed so not changed.`);
-		} else {
-			// Only insert the group and its children if the group did not already exist.
-			await group.insert(conn);
-			// If meterData[7] is not undefined then use value to set meter id.
-			// This is normally only done for the website data.
-			// It is best/easiest to do this before there are any members of the group.
-			const newId = groupData[7];
-			let parent;
-			if (newId != undefined) {
-				console.log('              group id set to ', newId);
-				group.id = newId;
-				const query = `update groups set id = ${group.id} where name = '${groupName}'`;
-				await conn.none(query);
-				parent = group;
-			} else {
-				// Get it again so have id.
-				parent = await Group.getByName(group.name, conn);
+		let ok = true;
```

Comment:
could you add comments to explain this snippet here? same for 476

Reply:
I added more comments in both places. Thanks for the suggestion.